plugins/generator-1.14.4/forge-1.14.4/procedures/world_data_num_setgamerule.java.ftl

```diff
@@ -0,0 +1,17 @@
+if (!world.isRemote()) {
+	MinecraftServer server = Objects.requireNonNull(world.getWorld().getServer());
+	server.getCommandManager().handleCommand(
+			new CommandSource(
+					server,
+					new Vec3d(0, 0, 0),
+					new Vec2f(0, 0),
+					(ServerWorld) world.getWorld(),
+					4,
+					"server",
+					new StringTextComponent("server"),
+					server,
+					null
+			),
+			String.format("gamerule %s %d", (${generator.map(field$gamerulesnumber, "gamerulesnumber")}).toString(), ${input$gameruleValue})
```

Comment:
are you sure this is supposed to be `toString`?

Reply:
For 1.14.4? I'm pretty sure yes, why? The `getName` method doesn't exist in 1.14.4

Reply:
`toString` is an inherited method from `Object` which `Gamerule.RuleKey` probably does not override. 
It may not return what you need it to return. Can you test this?

Reply:
apparently `RuleKey` overrides it, but it returns the value of the rule, not its name

Reply:
I tested it and it works fine.

Reply:
nvm

Reply:
actually hold on there
use `func_223576_a` instead

Reply:
What do I replace by `func_223576_a`?

Reply:
For 1.15.2, you could set int by doing this:

```java
world.getWorld().getGameRules().get(${generator.map(field$gamerulesboolean, "gamerulesboolean")}).func_223553_a("" + ${input$gameruleValue});
world.getWorld().getGameRules().get(${generator.map(field$gamerulesboolean, "gamerulesboolean")}).func_223556_a(world.getWorld().getServer());
```

Note the code is untested, but try from here. For 1.14.4 it should be similar.

Reply:
No, the code won't work because you're using "gameruleboolean", for gamerules that are integers, you need to use "gamerulesnumber"

Reply:
I said it is an untested code. I just checked methods used by commands to do numbers. If commands can do it, they can be avoided. Change mappings and it should be a good starting point.

Reply:
Yes, I'll try your code. Wdym by "change mappings"? It's to avoid users to select boolean gamerules in a procedure block that uses integer gamerule...

Reply:
I meant change "gamerulesboolean" to "gamerulesnumber"

Reply:
I tested your code and it doesn't work because it can't find `func_223553_a` and `func_223556_a`
This PR has been moved here : [https://github.com/MCreator/MCreator/pull/158](url)